lib/rules/no-array-prototype-extensions.js

```diff
@@ -424,28 +424,36 @@ function applyFix(callExpressionNode, fixer, context, options = {}) {
         // default to `compare` if the `compare` hasn't already been imported.
         const importedGetName = options.importedGetName ?? 'get';
 
-        const compareValueCheckText = `const compareValue = ${importedCompareName}(${importedGetName}(a, key), ${importedGetName}(b, key));
-            if (compareValue) {
-              return compareValue;
-            }`;
         const argsText = callArgs.map((arg) => sourceCode.getText(arg)).join(', ');
 
-        // Loop through keys if more than one of them.
-        const sortFn =
-          callArgs.length === 1
-            ? `(a, b) => {
-            const key = ${argsText};
-            ${compareValueCheckText}
+        let sortFn;
+
+        if (callArgs.length === 1 && callArgs[0].type !== 'SpreadElement') {
+          const comparisionFnInString = `${importedCompareName}(${importedGetName}(a, ${argsText}), ${importedGetName}(b, ${argsText}))`;
+
+          sortFn =
+            callArgs[0].type === 'CallExpression'
```

Comment:
I think it would be safer to flip this check. Check for `Literal` or `Identifier` in order to use the value directly. Otherwise, store it in `key`.